src/shutdown/umount.c

```diff
@@ -651,6 +651,17 @@ static int umount_with_timeout(MountPoint *m, int umount_log_level) {
                 if (r == -EBUSY)
                         log_umount_blockers(umount_log_level, m->path);
 
+                /* If API filesystems under /oldroot cannot be unmounted we can still lazily unmount
+                 * them to unblock /oldrood. They serve no function to us anymore and should be
+                 * memory-only and hence safe to unmount like this. */
+                if (r == -EBUSY && in_initrd() &&
```

Comment:
given it's a critical path, shouldn't this be in the previous r == -EBUSY block, and have log_umount_blockers as an else branch? No point in going through open files if the lazy one succeeds for these, no?

Reply:
I think there is still value in reporting any processes blocking these even if we then lazily unmount them. (Though, in this particular case there are no such processes as it's the nvidia kernel module itself blocking it somehow.)

Reply:
The issue is that there will be a cost to do those iterations, and also if all works out fine, but there are scary logs anyway, a lot of users would freak out unnecessarily. What about logging here if the lazy umount fails?

Reply:
That's what `umount_log_level` is for. It starts out at LOG_INFO and the blockers aren't logged until the log level is lower than that. It's only raised to LOG_ERR for a final umount round if we have no initrd to switch to (either because there is none or because we are in it). So it shouldn't add any costs.

It probably makes sense to change that param to a bool as it doesn't really reflect the regular log level to make it more clear.

Unless I got something wrong, we never even look for unmount blockers unless we are desperate, at which point we really should look and report for them.